go/libraries/doltcore/env/actions/commit.go

```diff
@@ -61,14 +132,14 @@ func GetNameAndEmail(cfg config.ReadableConfig) (string, string, error) {
 	return name, email, nil
 }
 
-func CommitStaged(ctx context.Context, dEnv *env.DoltEnv, props CommitStagedProps) error {
+func CommitStaged(ctx context.Context, ddb *doltdb.DoltDB, rsr env.RepoStateReader, rsw env.RepoStateWriter, props CommitStagedProps) (string, error) {
```

Comment:
I'd create a combined interface

```type RepoStateReadWriter interface {
    RepoStateReader
    RepoStateWriter
}```

and pass that in here rather than having two arguments

Reply:
I actually disagree with this. I think we should be explicit about which methods are only doing the reading and which are only doing the writing. I think a combined interface would obfuscate it.